src/main/java/org/elasticsearch/action/admin/indices/optimize/TransportOptimizeAction.java

```diff
@@ -121,11 +121,11 @@ protected GroupShardsIterator shards(ClusterState clusterState, OptimizeRequest
 
     @Override
     protected ClusterBlockException checkGlobalBlock(ClusterState state, OptimizeRequest request) {
-        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA);
+        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
```

Comment:
I don't think optimize should be `WRITE`.  It is an administrative type action, you aren't actually passing data, just telling the system to do something.